packages/react-magma-dom/src/components/ToggleButton/ToggleButton.tsx

```diff
@@ -0,0 +1,231 @@
+  /**
+   * Sets a disabled state for a button.
+   */
+  disabled?: boolean;
+  hasLabel?: boolean;
+  /**
+   * Sets a single selection group.
+   */
+  singleSelect?: boolean;
+  /**
+   * Icon which displays alongside text.
+   */
+  icon: React.ReactElement<IconProps>;
+  isChecked?: boolean;
+  isDefault?: boolean;
+  isInverse?: boolean;
+  /**
+   * Restricts deselection of a button within a single selection group.
+   */
+  requiredSelect: boolean;
+  /**
+   * Changes the button size throughout the group between 'small', 'medium', and 'large'.
+   */
+  size?: ButtonSize;
+}
+
+export type ToggleButtonProps = XOR<
+  ToggleButtonTextProps,
+  ToggleButtonIconProps
+>;
+
+//Sets the icon width for icon only Toggle Buttons.
+export function setIconWidth(props: ToggleButtonProps) {
+  if (props.size === ButtonSize.small) {
+    return props.theme.spaceScale.spacing07;
+  }
+  if (props.size === ButtonSize.medium) {
+    return props.theme.spaceScale.spacing09;
+  }
+  if (props.size === ButtonSize.large) {
+    return props.theme.spaceScale.spacing11;
+  }
+  return props.theme.spaceScale.spacing09;
+}
+
+//Sets the background colors for the Toggle Buttons.
+export function setBackgroundColor(props: ToggleButtonProps) {
+  if (
+    (props.isChecked && props.defaultChecked) ||
+    (!props.isChecked && props.isDefault)
+  ) {
+    return '';
+  }
+  if (props.isChecked || props.defaultChecked || props.isDefault) {
+    if (props.isInverse) {
+      return transparentize(0.5, props.theme.colors.neutral900);
+    }
+    return transparentize(0.5, props.theme.colors.neutral300);
+  }
+}
+
+//Background colors for the different button states.
+export const ToggleButtonStyles = props => css`
+  background: ${setBackgroundColor(props)};
+  &:not(:disabled):hover {
+    background: ${setBackgroundColor(props)};
+  }
+  &:not(:disabled):focus {
+    background: ${setBackgroundColor(props)};
+  }
+  &:hover {
+    background: ${setBackgroundColor(props)};
+  }
+`;
+const StyledToggleButtonIcon = styled(IconButton)`
+  ${ToggleButtonStyles}
+  min-width: auto;
+  width: ${props => (!props.hasLabel ? setIconWidth : '')};
+  span {
+    padding: ${props => (!props.hasLabel ? 0 : '')};
+  }
+`;
+
+const StyledToggleButtonText = styled(Button)`
+  ${ToggleButtonStyles}
+`;
+
+export const ToggleButton = React.forwardRef<
+  HTMLButtonElement,
+  ToggleButtonProps
+>((props, ref) => {
+  const {
+    'aria-label': ariaLabel,
+    children,
+    defaultChecked,
+    disabled,
+    icon,
+    isInverse,
+    testId,
+  } = props;
+  const { isDefault, requiredSelect, singleSelect } = React.useContext(
+    ToggleButtonGroupContext
+  );
+
+  const context = React.useContext(ToggleButtonGroupContext);
+
+  const theme = React.useContext(ThemeContext);
+
+  const [isChecked, setChecked] = React.useState(false);
+
+  const handleClick = () => {
+    if (isChecked) {
+      setChecked(false);
+    } else if (!isChecked) {
+      setChecked(true);
+    }
```

Comment:
When adopters use this, they are not only going to expect the button to change visually, but the buttons should likely trigger some logic as well. We need to add a call to their own button click function here. Also make sure to add an example of this in the docs/storybook to test it.